src/Background/stat_estimators.jl

```diff
@@ -1,5 +1,42 @@
 using Statistics
 
+"""
+    sigma_clip(data, sigma_low, sigma_high; center=median, std=std)
+    sigma_clip(data, sigma; center=median, std=std)
+    sigma_clip!(data, sigma_low, sigma_high; center=median, std=std)
+    sigma_clip!(data, sigma; center=median, std=std)
+
+This function returns sigma clipped values of the input `data`. The funtion `sigma_clip!` is an inplace implementation and mutates the `data`.
+`sigma_high` and `sigma_low` are for un-symmetrical clipping, when `sigma_low = sigma_high` then they can be passed as `sigma`.
+`center` and `std` are optional parameters which are functions for finding central element and standard deviation.
+
+# Example
+```jldoctest
+julia> data = [1, 2, 3];
+
+julia> sigma_clip(data, 1)
+3-element Array{Float64,1}:
+ 1.0
+ 2.0
+ 3.0
+
+julia> sigma_clip(data, 1, 1)
+3-element Array{Float64,1}:
+ 1.0
+ 2.0
+ 3.0
+```
+"""
```

Comment:
Please break this doc string up. I like to follow the pattern:

```julia
"""
    my_method!()

In-place version of [`my_method`](@ref). 
"""
my_method!()

"""
    my_method()

-Normal Full Docs`
"""
my_method() = my_method!(copy())
```.

I would also add an `!!! warning` admonition to the docs for `sigma_clip!` about how it can fail if types can't be coerced. RE our chat on slcak.

Reply:
Also, I think a much more explicit example would be
```julia
julia> x = randn(100);

julia> extrema(x)
(-2.60394004685248, 2.3098676948608743)

julia> sigma_clip!(x, 1); # clips to 1

julia>extrema(x)
(-1.0, 1.0)
```

( and make sure to add `using Random; Random.seed!(123456)` to the doctestsetup meta for that file. If that doesn't make sense I can show you exactly where.